packages/replay/src/integration.ts

```diff
@@ -192,6 +192,11 @@ Sentry.init({ replaysOnErrorSampleRate: ${errorSampleRate} })`,
     // Client is not available in constructor, so we need to wait until setupOnce
     this._loadReplayOptionsFromClient();
 
+    if (!this._options.sessionSampleRate && !this._options.errorSampleRate) {
```

Comment:
m: As discussed offline, let's not check for `0` here, but if the values have actually not been set - to account for the case when someone actually inits it with two `0` values.

Reply:
Done in [095dd92](https://github.com/getsentry/sentry-javascript/pull/6959/commits/095dd922b466119900de0ea23fabef25dd7f531f)

I had to do some initial refactoring to distinguish between default values and unset values so that we can correctly decide when to warn.